### ISSUE TITLE
BUG: pandas Parquet metadata is only being written to the _metadata file 

### ISSUE THREAD

**Issue by wesm** [dask/dask#2440]
Encountered as part of https://github.com/dask/dask/issues/2439

Setup:

```
import pandas as pd
import dask.dataframe as dd
from dask.dataframe.io.parquet import read_parquet, to_parquet

df = pd.DataFrame({'x': [6, 2, 3, 4, 5],
                   'y': [1.0, 2.0, 1.0, 2.0, 1.0]},
                  index=pd.Index([10, 20, 30, 40, 50], name='myindex'))
ddf = dd.from_pandas(df, npartitions=3)
to_parquet('example', ddf)
```

And then:

```
In [10]: import pyarrow.parquet as pq

In [11]: pq.ParquetFile('example/part.0.parquet').metadata.metadata
Out[11]: {}

In [12]: pq.ParquetFile('example/_metadata').metadata.metadata
Out[12]: {b'pandas': b'{"index_columns": ["myindex"], "columns": [{"name": "myindex", "pandas_type": "int64", "numpy_dtype": "int64", "metadata": null}, {"name": "x", "pandas_type": "int64", "numpy_dtype": "int64", "metadata": null}, {"name": "y", "pandas_type": "float64", "numpy_dtype": "float64", "metadata": null}], "pandas_version": "0.20.2"}'}
```

It seems OK to override the partition metadata with the `_metadata` file (we can handle this on the Arrow side for directory reads), but it may be a good idea to replicate the pandas metadata to all of the individual data set pieces

**Comment by wesm** [comment]
See https://issues.apache.org/jira/browse/ARROW-1103

**Comment by martindurant** [comment]
Good idea, metadata should be in every data file (as the parquet schema already is)

**Comment by wesm** [comment]
@martindurant there is also some implementation slippage, the parameter for the NumPy type is `numpy_type` per:

https://github.com/pandas-dev/pandas/blob/master/doc/source/developer.rst

**Comment by wesm** [comment]
We should probably set up an integration test repo someplace that is independent from dask, to make these issues easier to test. What do you think?

**Comment by martindurant** [comment]
You mean a fastparquet/arrow/other-parquet cross-library integration tests? That sounds like a decent idea, but of course someone would have to put effort into it. One might argue that dask's needs are more focussed.

**Comment by wesm** [comment]
I should think that Dask users would be concerned with correctness at the single Parquet file level. The intent would be to consolidate our unit testing efforts to put all "problematic pandas data" in one place so we can verify that data written by one implementation can be accurately read by another. I am willing to spend some time with setting up a repo for this under pandas-dev

**Comment by martindurant** [comment]
Sorry, accidental auto-close, dealing with numpy_type only.

**Comment by jcrist** [comment]
This issue (writing pandas metadata to all files) should be fixed in #2868.
